sorc/build_all.sh

```diff
@@ -82,19 +64,95 @@ while getopts ":a:dfghj:kuvwy" option; do
+# shellcheck disable=SC2034
+gfs_builds="gfs gfs_utils ufs_utils upp ww3_gfs"
+# shellcheck disable=SC2034
+gefs_builds="gefs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+sfs_builds="sfs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+gsi_builds="gsi_enkf gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+gdas_builds="gdas gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+all_builds="gfs gfs_utils ufs_utils upp ww3_gfs gefs ww3_gefs gdas gsi_enkf gsi_monitor gsi_utils"
+
 logs_dir="${HOMEgfs}/sorc/logs"
 if [[ ! -d "${logs_dir}" ]]; then
   echo "Creating logs folder"
   mkdir -p "${logs_dir}" || exit 1
 fi
 
-# Check final exec folder exists
-if [[ ! -d "${HOMEgfs}/exec" ]]; then
-  echo "Creating ${HOMEgfs}/exec folder"
-  mkdir -p "${HOMEgfs}/exec"
+# Jobs per build ("min max")
+declare -A build_jobs build_opts build_scripts
+build_jobs=(
+    ["gfs"]=8 ["gefs"]=8 ["sfs"]=8 ["gdas"]=8 ["gsi_enkf"]=2 ["gfs_utils"]=1 ["ufs_utils"]=1
+    ["ww3_gfs"]=1 ["ww3_gefs"]=1 ["gsi_utils"]=1 ["gsi_monitor"]=1 ["gfs_utils"]=1 ["upp"]=1
+)
+
+# Establish build options for each job
+_gfs_exec="gfs_model.x"
+_gefs_exec="gefs_model.x"
+_sfs_exec="sfs_model.x"
+build_opts=(
+    ["gfs"]="${wave_opt} ${_build_ufs_opt} ${_verbose_opt} ${_build_debug} -e ${_gfs_exec}"
+    ["gefs"]="${wave_opt} ${_build_ufs_opt} ${_verbose_opt} ${_build_debug} -e ${_gefs_exec}"
+    ["sfs"]="${wave_opt} ${_build_ufs_opt} ${_verbose_opt} ${_build_debug} -e ${_sfs_exec}"
+    ["upp"]="${_build_debug}"
+    ["ww3_gfs"]="${_verbose_opt} ${_build_debug}"
+    ["ww3_gefs"]="-w ${_verbose_opt} ${_build_debug}"
+    ["gdas"]="${_verbose_opt} ${_build_debug}"
+    ["ufs_utils"]="${_verbose_opt} ${_build_debug}"
+    ["gfs_utils"]="${_verbose_opt} ${_build_debug}"
+    ["gsi_utils"]="${_verbose_opt} ${_build_debug}"
+    ["gsi_enkf"]="${_verbose_opt} ${_build_debug}"
+    ["gsi_monitor"]="${_verbose_opt} ${_build_debug}"
+)
+
+# Set the build script name for each build
+build_scripts=(
+    ["gfs"]="build_ufs.sh"
+    ["gefs"]="build_ufs.sh"
+    ["sfs"]="build_ufs.sh"
+    ["gdas"]="build_gdas.sh"
+    ["gsi_enkf"]="build_gsi_enkf.sh"
+    ["gfs_utils"]="build_gfs_utils.sh"
+    ["ufs_utils"]="build_ufs_utils.sh"
+    ["ww3_gfs"]="build_ww3prepost.sh"
+    ["ww3_gefs"]="build_ww3prepost.sh"
+    ["gsi_utils"]="build_gsi_utils.sh"
+    ["gsi_monitor"]="build_gsi_monitor.sh"
+    ["gfs_utils"]="build_gfs_utils.sh"
+    ["upp"]="build_upp.sh"
+)
+
+# Check the requested systems to make sure we can build them
+declare -A builds
+system_count=0
+for system in "${@}"; do
+   # shellcheck disable=SC2076
+   if [[ " ${supported_systems[*]} " =~ " ${system} " ]]; then
+      (( system_count += 1 ))
+      build_list_name="${system}_builds"
+      for build in ${!build_list_name}; do
+         builds["${build}"]="yes"
+      done
+   else
+      echo "Unsupported build system: ${system}"
+      _usage
+   fi
+done
```

Comment:
@aerorahul Here is where the magic happens.

Reply:
excellent.  Thanks for pointing this out.